### DISCUSSION TITLE
Bookingcodes csv email header

### DISCUSSION THREAD

**Discussion by datengraben** [wielebenwir/commonsbooking#1264] [Questions & Answers]
1. Welchen Use-Case bedient das Feature der erzeugten und via E-Mail versendeten Buchungs-Codes?
    Mein Verständnis:
      * Direktes Drucken aus dem E-Mail Client
      * Öffnen der Datei aus dem Email-Client, dann drucken

    Ist das vollständig?

2. Reicht die aktuelle Implementierung mit der gewählten Kombination von CSV-Delimiter, Content-Type und Content-Transfer-Encoding dafür aus?

Im PR dafür hat @printpagestopdf in https://github.com/wielebenwir/commonsbooking/pull/1174#discussion_r1161272950_ einen Vorschlag gemacht, den ich hier weiter diskutieren wollte.

**Comment by datengraben** [comment]
Meine Frage an @printpagestopdf aus der PR Diskussion: Was genau ist deiner Meinung nach das Problem mit Content-Type: text/csv und Content-Transfer-Encoding binary falsch? Ein Nachteil für UTF8 Text mit Tabulatoren ergibt sich doch nicht, aber vielleicht habe ich Transfer-Encoding auch falsch verstanden. Ich habe es so verstanden das 7bit und 8bit Transfer-Encodings das Zeilenlängen-Limit nicht ausschließen und man so aufpassen muss nicht mehr als 1000 Zeichen pro Zeile zu übertragen. Also können wir doch theoretisch 8bit oder binary nehmen.

**Reply by printpagestopdf** [reply]
Hm, also Content-Type: text/csv und Content-Transfer-Encoding binary finde ich vollständig richtig. Auslösende Frage von @hansmorb war ja, ob die Fileextension .txt passend ist. Meiner Meinung nach hängt das vom primären Use Case ab. Wenn dieser bedeutet möglichst einfach ausdrucken zu können, dann würde ich die Fileextension .txt lassen. Wenn wir aber davon ausgehen, dass eher eine vorgeschaltete Bearbeitung in einem office Programm stattfinden soll, dann würde ich nicht nur die Fileextension auf .csv ändern sondern auch den Fieldseparator. Da ich den ursprünglichen Use Case auch nicht kenne kann ich leider statt einer Antwort nur diese Frage zurückgeben.
